pkg/cloud/api/mutable_resource.go

```diff
@@ -195,11 +195,41 @@ func (u *mutableResource[GA, Alpha, Beta]) CheckSchema() error {
 	if err != nil {
 		return err
 	}
+	ga, _ := u.ToGA()
+	alpha, _ := u.ToAlpha()
+	err = checkSubsetOf(ga, alpha)
+
+	if err != nil {
+		return fmt.Errorf("checkSubsetOf(ga, alpha) = %v, want nil", err)
+	}
+
 	err = checkSchema(reflect.TypeOf(&u.beta))
 	if err != nil {
 		return err
 	}
-	// TODOD(kl52752) Add validation that GA is a subset of Beta and Alpha.
+	beta, _ := u.ToBeta()
+	err = checkSubsetOf(ga, beta)
+
+	if err != nil {
+		return fmt.Errorf("checkSubsetOf(ga, beta) = %v, want nil", err)
+	}
+
+	return nil
+}
+
+func checkSubsetOf[T1 any, T2 any](t1 *T1, t2 *T2) error {
+	if isPlaceholderType(t2) {
```

Comment:
Don't put this logic here. It's confusing and technically placeholder type is not a superset of any struct except the empty struct. This will lead to bugs later because the special case behavior is not consistent.

Reply:
done